### ISSUE TITLE
Master branch failed to build the debug version

### ISSUE THREAD

**Issue by Leslie-Fang** [tensorflow/tensorflow#37498]
**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Centos7.6
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary): source 
- TensorFlow version: master, commit-id: b45adaf6efeeb8e4acf8517a01f7dc01bdf21db9
- Python version: 3.6
- Installed using virtualenv? pip? conda?:
- Bazel version (if compiling from source): bazel 2.0.0
- GCC/Compiler version (if compiling from source): 8.3

**Describe the problem**
The debug version build failed with the command:
```
bazel build -c dbg //tensorflow/tools/pip_package:build_pip_package
```
Error:
```
external/aws-checksums/source/intel/crc32c_sse42_asm.c: In function 's_crc32c_sse42_clmul_256':
external/aws-checksums/source/intel/crc32c_sse42_asm.c:61:5: error: 'asm' operand has impossible constraints
     asm volatile("enter_256_%=:"
     ^~~
Target //tensorflow/tools/pip_package:build_pip_package failed to build
```

**Comment by Saduf2019** [comment]
@Leslie-Fang 
could you please downgrade the GCC/Compiler version to 7.x and build.

**Comment by Leslie-Fang** [comment]
Thanks @Saduf2019 Since I don't have GCC7.x available. I have tried GCC63 which gives the same error.
Is GCC7.x specifically required to build the debug version?

**Comment by pw2393** [comment]
Hi @Saduf2019, I'm using `gcc==7.5.0` to build with the above command in debug mode, it raises the same issue.

**Comment by mikeurbach** [comment]
i'm running into this on gcc 7.5.0 as well. i was able to work around this by adding a small change in here https://github.com/tensorflow/tensorflow/blob/master/third_party/aws/aws-checksums.bazel#L24. below the various options that are already there, i added
```
    defines = [                                                                                                                                                                                                    
        "DEBUG_BUILD"                                                                                                                                                                                              
    ]                                                                                                                                                                                                              
```
that seemed to do the trick. the advice came from this comment https://github.com/awslabs/aws-checksums/issues/8#issuecomment-440027793.

hope this helps someone else. i'm happy to help try to figure out or test a fix for this issue.

**Comment by bondhugula** [comment]
I still experience this issue with fa0721cfbd93a1506d39735296a260a877354e6c (May 21). The real issue here is that the inline asm code in question isn't meant to compile with debug build flags (-c dbg), but only with -O3/-O2/-O1 specified. So, aws-checksums uses DEBUG_BUILD to avoid that inline asm and use an alternative slow implementation for debug builds. As such, for TensorFlow's debug builds, one should pass -DDEBUG_BUILD to aws-checksums so that the inline asm code path is avoided (it has hardcoded register constraints). For more information:
https://github.com/awslabs/aws-checksums/issues/8
The workaround suggested in the post above works, but that would use the slow implementation even under release builds. Instead, DEBUG_BUILD for aws-checksums should be defined only under bazel debug build config. fastbuild and opt conf should be fine without that.

**Comment by jgehw** [comment]
I provided a fix in PR #42743 which adds DEBUG_BUILD only for linux debug builds
